### PR TITLE
Summer update!

### PR DESCRIPTION
**Pull request by ShadzFen**
## About The Pull Request

**IF YOU ARE GOING TO REVIEW THIS PR, PLEASE READ THE FOLLOWING:**
**I did as much spellchecking as possible, but I'm utterly useless at it, so PLEASE make sure to check the spelling on all names and descriptions while reviewing and point out anything that needs changing while the PR is up.**

Right, now that that's out of the way... SUMMER UPDATE!!!!!

This PR is gonna be a big one, so brace yourselves...
I've added a large selection of brand new milkshakes for us all to enjoy, as well as a few new fruits and their respective fruit juices too! Some condiments are added as well. All milkshake, plant and fruit spritework is done by Sigmasquad, so go say a BIG THANK YOU to him. He did an outstanding job. We got new juice sprites too!

**NEW FRUITS, CONDIMENTS AND PLANTS**
- Blueberries
- Strawberries
- Pineapples
- Cinnamon
- Mint.

image
image
image
image

**NEW FRUIT JUICES**
- New: Blueberry juice
- New: Pineapple juice
- New: Strawberry juice
- New: Jungle juice.
- New: Pumpkin juice
- Updated Sprite: Grape juice
- Updated Sprite: Watermelon juice

image

**NEW CONDIMENTS**
- Blended Mint
- Cinnamon Powder

**NEW COCKTAILS AND OTHERS**
- Mojito
- Mint Chocky

**NEW MILKSHAKES**
- Blueberry milkshake
- Pineapple milkshake
- Strawberry milkshake
- Chocolate milkshake
- Mint Chocolate milkshake
- Banana milkshake
- Berry milkshake
- Grape milkshake
- Watermelon milkshake
- Cherry milkshake
- Pumpkin Spice milkshake
- Jungle milkshake

image image
image
image

The milkshake sprites have a New() proc that allows them to randomly pick between 4 different sprites on creation. They can either be a simple milkshake, have a straw, have an umbrella, or have both! This should help give them even more variety.

Every single milkshake has a recipe which has been tested and works without getting in the way of other drinks. Everything can be made from round start by simply planting and gathering the necessary fruits, condiments and drinks to mix and blend.

A few small changes were also made, such as fixing the pumpkin sprite so it actually displayed a pumpkin rather than broken ass white melon.
image

It can also now be blended to make pumpkin juice, which is kinda bland and not that nice to drink, but it's used as a condiment to make a much nicer pumpkin spice milkshake, and possibly in the future to make a pumpkin spice latte for a potential autumn update.

I can't think of anything else so... Here's the last part of the PR. These are the DMI changes. **All sprites are NEW, no existing sprites have been modified.**

image
image
image
image

Time to properly enjoy the summer! 

## Why It's Good For The Game

It might be a little late, but I wanted to make a summer update because I love summer, and I want other people to enjoy it too! Yes, even the people who hate the heat. That's why we've got all these new refreshing juices and milkshakes! Something nice to enjoy by the new pool we added.

## Changelog
```changelog
add: A large selection of milkshakes that you can find on the Neon Cocktails app on your PDA! List is too large to post here.
add: A selection of new fruits and condiments that can be found at the very bottom of the list of seeds from the seed vendomat.
add: New fruit juices for everyone to enjoy based off the new fruits that were added. Some old juice sprites were updated too.
add: Mojito and Mint Chocky
fix: Pumpkin sprite now displays properly.
```

### CODE REVIEW

**Comment by ShadzFen** [comment]
Oh, I also forgot to mention that the new modular DM files explain in a fair amount of details HOW all of the plant, fruit and seed system works. It took me a while to figure it out, so I wanted to make sure it was clear for anyone new who might want to add any seeds, fruits, plants, etc. of their own!

File: zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+	glass_desc = "A very fruity tropical flavour with a mix of sour and sweet fruit juices, cooled to perfection in order to quench the thirst of a hot summer day!"
+
+
+
+
+///////MILKSHAKES///////
+
+/datum/reagent/drink/milkshake/blueberry
+	name = "Blueberry Milkshake"
+	description = "A sweet blueberry flavour with a smooth blend of cream and milk, shaked and cooled to perfection in order to guarantee palate satisfaction, all topped with a light layer of froth."
```
Comment (Michiyamenotehifunana) [inline zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm:88]: Past tense of shake is shaken, not shaked.

File: zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+	taste_description = "thick creamy blueberry"
+	id = "blueberry_milkshake"
+	color = "#969CE5"
+	adj_temp = -9
+	taste_tag = list(TASTE_LIGHT, TASTE_SWEET, TASTE_SLIMEY)
+
+	glass_icon_state = "shake_blueberry"
+	glass_name = "Blueberry Milkshake"
+	glass_desc = "A sweet blueberry flavour with a smooth blend of cream and milk, shaked and cooled to perfection in order to guarantee palate satisfaction, all topped with a light layer of froth."
```
Comment (Michiyamenotehifunana) [inline zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm:97]: Ditto.

File: zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+	glass_center_of_mass = list("x"=16, "y"=7)
+
+/datum/reagent/drink/milkshake/blueberry/New()
+	var/toppings = pick("_solo", "_straw", "_umbr", "_both")
+	glass_icon_state += toppings
+
+/datum/reagent/drink/milkshake/pineapple // We have no pineapple fruits.
+	name = "Pineapple Milkshake"
+	description = "A sweet and strong pineapple flavour with a smooth blend of cream and milk, shaked and cooled to perfection in order to guarantee palate satisfaction, all topped with a light layer of froth."
```
Comment (Michiyamenotehifunana) [inline zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm:106]: Uhhh, should use dreamcrasher's find and replace function to replace all instances of "shaked" with "shaken".

File: zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+	glass_desc = "A sweet and gentle banana flavour with a smooth blend of cream and milk, shaked and cooled to perfection in order to guarantee palate satisfaction, all topped with a light layer of froth."
+	glass_center_of_mass = list("x"=16, "y"=7)
+
+/datum/reagent/drink/milkshake/banana/New()
+	var/toppings = pick("_solo", "_straw", "_umbr", "_both")
+	glass_icon_state += toppings
+
+/datum/reagent/drink/milkshake/berry
+	name = "Berry Milkshake"
+	description = "A sweet forrest berry flavour with a smooth blend of cream and milk, shaked and cooled to perfection in order to guarantee palate satisfaction, all topped with a light layer of froth."
```
Comment (Michiyamenotehifunana) [inline zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm:196]: Forrest with two Rs is a given name, forest refers to the large tracts of trees.

File: zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+	taste_description = "fruity creamy berries"
+	id = "berry_milkshake"
+	color = "#B76B99"
+	adj_temp = -9
+	taste_tag = list(TASTE_LIGHT, TASTE_SWEET, TASTE_SLIMEY)
+
+	glass_icon_state = "shake_berry"
+	glass_name = "Berry Milkshake"
+	glass_desc = "A sweet forrest berry flavour with a smooth blend of cream and milk, shaked and cooled to perfection in order to guarantee palate satisfaction, all topped with a light layer of froth."
```
Comment (Michiyamenotehifunana) [inline zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm:205]: Ditto.

File: zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+	glass_desc = "A sweet and strong pineapple flavour with a smooth blend of cream and milk, shaked and cooled to perfection in order to guarantee palate satisfaction, all topped with a light layer of froth."
+	glass_center_of_mass = list("x"=16, "y"=7)
+
+/datum/reagent/drink/milkshake/pineapple/New()
+	var/toppings = pick("_solo", "_straw", "_umbr", "_both")
+	glass_icon_state += toppings
+
+/datum/reagent/drink/milkshake/strawberry // We have no strawberry fruits.
+	name = "Strawberry Milkshake"
+	description = "A sweet suggary strawberry flavour with a smooth blend of cream and milk, shaked and cooled to perfection in order to guarantee palate satisfaction, all topped with a light layer of froth."
```
Comment (Michiyamenotehifunana) [inline zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm:124]: Sugary has a single G only.

File: zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+	taste_description = "sweet creamy strawberry"
+	id = "strawberry_milkshake"
+	color = "#FE7F7F"
+	adj_temp = -9
+	taste_tag = list(TASTE_LIGHT, TASTE_SWEET, TASTE_SLIMEY)
+
+	glass_icon_state = "shake_strawberry"
+	glass_name = "Strawberry Milkshake"
+	glass_desc = "A sweet suggary strawberry flavour with a smooth blend of cream and milk, shaked and cooled to perfection in order to guarantee palate satisfaction, all topped with a light layer of froth."
```
Comment (Michiyamenotehifunana) [inline zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm:133]: Ditto.

**Review by Michiyamenotehifunana — COMMENTED** [review]
Just a quick lookthrough for spelling. Only 3 minor things that can be easily ctrl+h'd away.

File: zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm (hunk 4 above)
Comment (Jamini) [reply zzzz_modular_occulus/code/modules/reagents/reagents/food-Drinks.dm:196]: A

**Review by Jamini — CHANGES_REQUESTED** [review]
This branch is not fully brought up to master. As per the flavor code expansion (#625) seed datums *need* to have appropriate taste_tag  var lists in order to be appropriately considered in the construable food update! By default seeds will only have vegan and vegetarian flavors.

Additionally zzzz_modular_occulus/code/modules/cooking/seed.dm  should probably be merged with this file since it also modifies existing seed datums already.

**Comment by ShadzFen** [comment]
Changes made. Ready for review again.

**Review by Jamini — APPROVED** [review]
Everything looks good here. If anyone else wants to double-check spelling stuff that would be good. I'm not good at that.

**Comment by RadiantFlash7** [comment]
I'll double check spelling really quick

**Comment by RadiantFlash7** [comment]
> I'll double check spelling really quick

Spelling good. No issues I could see :3

**Pull request merged**